spec/services/csv/lettings_log_csv_service_spec.rb

```diff
@@ -1,430 +1,227 @@
+  let(:log) do
+    create(
+      :lettings_log,
+      :completed,
+      startdate: fixed_time,
+      created_at: fixed_time,
+      updated_at: fixed_time,
+      mrcdate: fixed_time - 1.day,
+      voiddate: fixed_time - 2.days,
+      propcode: "ABCDEFG",
+      tenancycode: "HIJKLMN",
+      postcode_full: "NW9 5LL",
+      ppostcode_full: "TN23 6LZ",
+      created_by: user,
+      managing_organisation: organisation,
+    )
+  end
+  let(:user) { create(:user, :support, email: "s.port@jeemayle.com") }
+  let(:service) { described_class.new(user:, export_type:) }
+  let(:export_type) { "labels" }
+  let(:csv) { CSV.parse(service.prepare_csv(LettingsLog.where(id: logs.map(&:id)))) }
+  let(:logs) { [log] }
+  let(:headers) { csv.first }
+
+  it "calls the form handler to get all questions in order when initialized" do
+    allow(FormHandler).to receive(:instance).and_return(form_handler_mock)
+    allow(form_handler_mock).to receive(:ordered_lettings_questions_for_all_years).and_return([])
+    service
+    expect(form_handler_mock).to have_received(:ordered_lettings_questions_for_all_years)
+  end
+
+  it "returns a string" do
+    result = service.prepare_csv(LettingsLog.all)
+    expect(result).to be_a String
+  end
+
+  it "returns a csv with headers" do
+    expect(csv.first.first).to eq "id"
+  end
+
+  context "when stubbing :ordered_lettings_questions_for_all_years" do
+    let(:lettings_form) do
+      FormFactory.new(year: 2050, type: "lettings")
+                 .with_sections([build(:section, :with_questions, question_ids:, questions:)])
+                 .build
+    end
+    let(:question_ids) { nil }
+    let(:questions) { nil }
 
     before do
-      LettingsLog.create!(startdate: Time.zone.today, created_at: Time.utc(2022, 2, 8, 16, 52, 15))
-      allow(FormHandler.instance).to receive(:get_form).and_return(real_2021_2022_form)
+      allow(FormHandler).to receive(:instance).and_return(form_handler_mock)
+      allow(form_handler_mock).to receive(:form_name_from_start_year)
+      allow(form_handler_mock).to receive(:get_form).and_return(lettings_form)
+      allow(form_handler_mock).to receive(:ordered_lettings_questions_for_all_years).and_return(lettings_form.questions)
+      allow(form_handler_mock).to receive(:lettings_in_crossover_period?).and_return false
+    end
+
+    context "when it returns questions with particular ids" do
+      let(:question_ids) { %w[prevten startdate brent rent_type] }
+
+      it "includes log attributes related to questions to the headers" do
+        expect(headers).to include(*question_ids.first(3))
+      end
+
+      it "removes some log attributes related to questions from the headers and replaces them with their derived values in the correct order" do
+        expect(headers).not_to include "rent_type"
+        expect(headers).to include(*%w[wrent renttype rent_type_detail])
+      end
     end
 
-    it "sets csv attributes in correct order" do
```

Comment:
Do we no longer want to test the order?

Reply:
it's tested implicitly in the full csv download tests